### DISCUSSION TITLE
session.stream(stmt) without generating the orm result objects even if the query is an orm query

### DISCUSSION THREAD

**Discussion by WilliamStam** [sqlalchemy/sqlalchemy#12054] [Usage Questions]
im making a utility class that takes any stmt and executes it and populates my own models with the values. how can i skip SA from generating a ORM row result?

ie

```
result = await conn.session.stream(query.execution_options(yield_per=1000))
async for row in result:
    print(row._mapping) 
```

a "cheap" fix i found is to wrap the query 

```
query = select("*").select_from(query.subquery())
```
but this breaks asyncmy at the moment and just feels wrong in anycase. surely there must be a way to tell SA to always return a "normal" row object in the generator? im on hour 4 of google + docs and im starting to die inside. 

----

**Comment by CaselIT** [comment]
Hi,

If you don't want orm instances back either select the table directly instead of the class or use a connection instead of a session

**Comment by WilliamStam** [comment]
in some places the select is on an orm in other places its text() etc. for both of those the values need to go into a model that gets passed to the utility. (consider where the select uses joins as well. its not a straight orm object back) 

when you do select(UserTable) then for row in result.scalars() right. but when any other way then it needs to be for row in result: 

for the "non orm" result then row._mapped works to get a dict. for the orm result then it needs to do a lot more stuff. also if im never going to use the orm stuff in this case then why have the overhead if not needed? 

it needs to be against a session cause need to issue some commands (like switch schema for oracle) before the query.

**Reply by CaselIT** [reply]
in some places the select is on an orm in other places its text() etc. for both of those the values need to go into a model that gets passed to the utility. (consider where the select uses joins as well. its not a straight orm object back)

Your best option is to use a connection then.

> also if im never going to use the orm stuff in this case then why have the overhead if not needed?

then why use the orm? use tables instead

> it needs to be against a session cause need to issue some commands (like switch schema for oracle) before the query.

a session uses a connection, so I don't see why you would require the execute to be on a session. You can get the connection the session is using with this method https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.connection
